src/csharp/Microsoft.Spark/RDD/Collector.cs

```diff
@@ -66,18 +66,10 @@ internal interface IDeserializer
         /// </summary>
         private sealed class BinaryDeserializer : IDeserializer
         {
-            [ThreadStatic]
-            private static MaxLengthReadStream s_slicedReadStream;
-
             private readonly BinaryFormatter _formater = new BinaryFormatter();
 
             public object Deserialize(Stream stream, int length)
             {
-                MaxLengthReadStream readStream = s_slicedReadStream ??
-                    (s_slicedReadStream = new MaxLengthReadStream());
-
-                readStream.Reset(stream, length);
```

Comment:
I have removed `MaxLengthReadStream` because it's not used.

As you can see in the line below the original stream is passed to the formatter, not the instance of `MaxLengthReadStream`.